src/main/java/com/rultor/agents/ec2/AmazonEC2.java

```diff
@@ -0,0 +1,139 @@
+ * with the distribution. 3) Neither the name of the rultor.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.rultor.agents.ec2;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.services.ec2.AmazonEC2Client;
+import com.amazonaws.services.ec2.model.DescribeInstanceStatusRequest;
+import com.amazonaws.services.ec2.model.DescribeInstanceStatusResult;
+import com.amazonaws.services.ec2.model.Instance;
+import com.amazonaws.services.ec2.model.InstanceState;
+import com.amazonaws.services.ec2.model.RunInstancesRequest;
+import com.jcabi.aspects.Immutable;
+import com.jcabi.aspects.Tv;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Amazon EC2 instance on demand.
+ *
+ * @author Marton Horvath (marton.horvath@m323.org)
+ * @version $Id$
+ * @since 2.0
+ */
+@Immutable
+public class AmazonEC2 implements Amazon {
+    /**
+     * AWS credentials.
+     */
+    private final transient AWSCredentials credentials;
+    /**
+     * AWS region.
+     */
+    private final transient String region;
+    /**
+     * EC2 instance type.
+     */
+    private final transient String itype;
+    /**
+     * EC2 instance key pair name.
+     */
+    private final transient String keyname;
+
+    /**
+     * Public ctor.
+     * @param cred Credentials
+     * @param zone AWS region
+     * @param type Instance type
+     * @param key Instance key pair name
+     * @checkstyle ParameterNumberCheck (12 lines)
+     */
+    public AmazonEC2(
+        @NotNull final AWSCredentials cred,
+        @NotNull final String zone,
+        @NotNull final String type,
+        @NotNull final String key) {
+        super();
+        this.credentials = cred;
+        this.region = zone;
+        this.itype = type;
+        this.keyname = key;
+    }
+
+    // @checkstyle MagicNumberCheck (30 lines)
+    @Override
+    public final Instance runOnDemand() throws IOException {
+        Instance instance = null;
```

Comment:
please don't create null instance, when unable to get instance, just throw `IllegalStateException` with description of what happened